owl_wms/nn/attn.py

```diff
@@ -50,51 +60,32 @@ def __init__(self, config : 'TransformerConfig'):
         self.rope = FlatVideoRoPE(config)
         #self.rope = FrameRoPE(config)
 
-        self.tokens_per_frame = config.tokens_per_frame
-        self.causal = config.causal
-        
-        self.mask = create_block_causal_mask(
-            self.tokens_per_frame*config.n_frames,
-            self.tokens_per_frame
-        )
-    
-    def forward(self, x, kv_cache = None):
+    @torch.compile
+    def forward(self, x, block_mask, kv_cache=None):
+        B, L, _ = x.shape
+
         qkv = self.qkv(x)
         qkv = qkv.view(qkv.shape[0], qkv.shape[1], 3, self.n_heads, -1)
         q, k, v = qkv.permute(2, 0, 3, 1, 4)
```

Comment:
Can you change this to an einops?

Reply:
no big deal if not.

Reply:
Making a note to do something like the below once I'm done testing
```
q, k, v = rearrange(qkv, "b t (3 h d) -> 3 b h t d", h=self.n_heads)
```